### PR TITLE
Rianfowler/sc 50084/fix multi app ux 2

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- converts AppDetailsPage into a stateless functional component 
- necessary to use new app hooks 

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/50084/fix-multi-app-ux

#### Does this PR introduce a user-facing change?
none- refactor

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/components/context/withTheme.jsx [hunk 1]
```diff
@@ -10,3 +10,13 @@ export default function withTheme(Component) {
     );
   };
 }
+
+function useTheme() {
```
Comment (rianfowler) [inline web/src/components/context/withTheme.jsx:14]: new custom hook for getting the theme

File: web/src/components/apps/AppDetailPage.tsx [hunk 2]
```diff
@@ -85,69 +90,120 @@ class AppDetailPage extends Component<Props, State> {
...
       requiredKotsUpdateMessage: "",
       selectedWatchName: "",
-    };
-  }
+    }
+  );
+
+  const history = useHistory();
+  const params = useParams<KotsParams>();
+  const [currentApp, setCurrentApp] = useState(
```
Comment (rianfowler) [inline web/src/components/apps/AppDetailPage.tsx:98]: i have a hook for the current app but i'm not using it here yet. I need to wire up the `useApps` stuff so the `currentApp` will be accurate. will do that in the next pr

File: web/src/Root.tsx [hunk 3]
```diff
@@ -66,7 +66,9 @@ const ProtectedRoute = ({
 };
 
 const ThemeContext = React.createContext({
-  setThemeState: () => {},
+  setThemeState: (themeState?: ThemeState) => {
+    console.log("setThemeState used before being set", themeState);
```
Comment (alicenstar) [inline web/src/Root.tsx:70]: Was this meant to be here? It's fine if so, just wanted to double check.

File: web/src/components/apps/AppDetailPage.tsx [hunk 4]
```diff
@@ -202,442 +258,387 @@ class AppDetailPage extends Component<Props, State> {
...
+                  exact
+                  path="/app/:slug/registry-settings"
+                  render={() => (
+                    <AppRegistrySettings
+                      app={app}
+                      updateCallback={refetchData}
+                    />
+                  )}
+                />
+                {true && (
```
Comment (alicenstar) [inline web/src/components/apps/AppDetailPage.tsx:562]: Is this true supposed to be hardcoded?

File: web/src/Root.tsx (hunk 3 above)
Comment (rianfowler) [reply web/src/Root.tsx:70]: Yeah this is just a placeholder for now. Going to move the themeset stuff into it's own provider hook and i wanted to make sure i remember to add some error handling.

File: web/src/components/apps/AppDetailPage.tsx (hunk 4 above)
Comment (rianfowler) [reply web/src/components/apps/AppDetailPage.tsx:562]: no- i added another pr to fix this.

**Pull request merged**